clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/GCEUtil.groovy

```diff
@@ -393,6 +396,43 @@ class GCEUtil {
     return ((tags ?: []) + securityGroupTags).unique()
   }
 
+  static Boolean shouldUpdateUrlMap(UrlMap urlMap, UpsertGoogleLoadBalancerDescription description) {
+    if (!urlMap) {
+      return false
+    }
+    def defaultServiceName = "${description.loadBalancerName}-${UpsertGoogleHttpLoadBalancerAtomicOperation.BACKEND_SERVICE_NAME_PREFIX}-${description.defaultService.name}"
+    Boolean defaultDiffers = getLocalName(urlMap.getDefaultService()) != defaultServiceName
+    Map<List<String>, Map> existingHostRuleMap = [:] // Nested: [sorted host pattern list -> [sorted url pattern list -> local backend service name]]
+
+    urlMap?.getHostRules()?.each { HostRule hostRule ->
+      urlMap?.getPathMatchers()?.each { PathMatcher pathMatcher ->
+        if (pathMatcher.getName() == hostRule.getPathMatcher()) {
+          Map pathRuleMap = [:]
+          List<PathRule> pathRules = pathMatcher.getPathRules()
+          pathRules?.each { PathRule pathRule ->
+            pathRuleMap.put(pathRule.getPaths().sort(), getLocalName(pathRule.getService()))
+          }
+          existingHostRuleMap.put(hostRule.getHosts().sort(), pathRuleMap)
+        }
+      }
+    }
+
+    Boolean mapServicesDifferences = false
+    description?.hostRules?.each { GoogleHostRule googleHostRule ->
+      def hostPatterns = googleHostRule.hostPatterns.sort()
+      googleHostRule?.pathMatcher?.pathRules?.each { GooglePathRule googlePathRule ->
+        def urlPatterns = googlePathRule?.paths?.sort()
+        def serviceName = "${description.loadBalancerName}-${UpsertGoogleHttpLoadBalancerAtomicOperation.BACKEND_SERVICE_NAME_PREFIX}-${getLocalName(googlePathRule.backendService.name)}"
+        Map urlPatternMap = existingHostRuleMap.get(hostPatterns)
+        if (!urlPatternMap || !urlPatternMap.containsKey(urlPatterns) || urlPatternMap.get(urlPatterns) != serviceName) {
+          mapServicesDifferences = true // Groovy, baby.
+          return
+        }
+      }
+    }
+    return defaultDiffers || mapServicesDifferences
```

Comment:
Looks like you can short circuit doing that large check for `mapServiceDifferences`.